### ISSUE TITLE
Clarify variables of BatchNorm*d functions

### ISSUE THREAD

**Issue by cowwoc** [pytorch/pytorch#68798]
## 📚 Documentation

https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html and BatchNorm for higher dimensions reference (but do not define) variables N,C,D,H,W.

I am expecting all variables to be described similar to how https://pytorch.org/docs/stable/generated/torch.nn.RNN.html does.

**Comment by jbschlosser** [comment]
As the author of the RNN docs, I'm glad you appreciate the explicit descriptions there :)

Being explicit like this is something we could consider applying globally across the docs. Up until now, the docs have relied on an implicit convention for dimension naming, with the recent, work-in-progress [style guide](https://github.com/pytorch/pytorch/wiki/torch.nn-Module-Documentation-Style-Guide) spelling out the convention.

I'll bring this up in internal discussion to see what others think.
